pkg/provisioner/crdprovisioner.go

```diff
@@ -229,10 +230,7 @@ func (c *CrdProvisioner) CreateVolume(
 				Secrets:                   secrets,
 				ContentVolumeSource:       volumeContentSource,
 				AccessibilityRequirements: accessibilityReq,
-			},
-			Status: diskv1beta1.AzVolumeStatus{
-				PersistentVolume: parameters[consts.PvNameKey],
-				State:            diskv1beta1.VolumeOperationPending,
+				PersistentVolume:          parameters[consts.PvNameKey],
```

Comment:
What guarantees do we have that the `consts.PvNameKey` is set in this map? If it's not there, we'll panic. We should fail gracefully if it is missing.

Reply:
https://kubernetes-csi.github.io/docs/external-provisioner.html#persistentvolumeclaim-and-persistentvolume-parameters

This gets added by the external-provisioner when `--extra-create-metadata` is set when installing the provisioner.

Reply:
Right. However, it would result in a panic if this were omitted in the `csi-provisioner`. Let's be defensive against this case and log an appropriate error.